components/base/wheeled/wheeled_base.go

```diff
@@ -372,16 +371,8 @@ func (wb *wheeledBase) SetVelocity(ctx context.Context, linear, angular r3.Vecto
 	// finish "spinning" have undefined behavior due to the motors actually running at a
 	// speed/direction that was not intended.
 	wb.opMgr.CancelRunning(ctx)
-	defer wb.mu.Unlock()
-	for _, m := range wb.left {
-		errs = append(errs, m.GoFor(ctx, leftRPM, numRevolutions, nil))
-	}
-
-	for _, m := range wb.right {
-		errs = append(errs, m.GoFor(ctx, rightRPM, numRevolutions, nil))
-	}
-
-	return multierr.Combine(errs...)
+	wb.mu.Unlock()
+	return wb.runAll(ctx, leftRPM, numRevolutions, rightRPM, numRevolutions)
```

Comment:
I think it would be good to make this function more generic so that it could be used in the stop and setPower, but I see what youre saying that you would have to add parameters for each function which seems not great. cant really think of a good solution besides maybe making a map or array of the parameters needed for the function you're calling since they are all floats, but that seems messy. If you keep it like this though I think  the name `runAll` could be changed to `runAllGoFor` to make it more clear.

Reply:
yeah I agree. for now will just make the name change because it's less messy